presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -532,8 +539,14 @@ private void advanceToNextStripe()
 
         StripeInformation stripeInformation = stripes.get(currentStripe);
         validateWriteStripe(stripeInformation.getNumberOfRows());
+        List<Slice> stripeDecryptionKeyMetadata = getDecryptionKeyMetadata(currentStripe);
+        if ((!stripeDecryptionKeyMetadata.isEmpty() && !currentGroupDwrfDecryptors.isPresent())
```

Comment:
It is quite hard to follow this condition. What is checked here?

Reply:
if there are encrytped columns and currentGroupDwrfDecryptors hasn't been set yet
or it has been set, but we have new encryption keys (that second condition was missing a `!`) 

Then we update currentGroupDwrfDecryptors for the new keys